### PR TITLE
opaeuio: add opae/uio.h

### PR DESCRIPTION
**Pull request by tswhison**
Correct python command lines in opae/vfio.h.

Signed-off-by: Tim Whisonant <tim.whisonant@intel.com>

### CODE REVIEW

File: include/opae/uio.h [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+/**
+ * @file opae/uio.h
+ * @brief APIs to manage a PCIe device via UIO.
```
Comment (r-rojo) [inline include/opae/uio.h:29]: Not sure if PCIe is accurate enough but I'm not sure of a better description either.
You do mention DFL later so I'm not sure if this API applies only to dfl-uio devices or any device bound to a uio driver. It seems to me that being part of the dfl driver is a requirement.

File: include/opae/uio.h [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+ * The device structure u must have been previously opened by a call to
+ * opae_uio_open.
+ *
+ * @param[in] u     The open OPAE UIO device.
+ * @param[in] index The zero-based index of the desired MMIO region.
+ * @param[out] ptr  Optional pointer to receive the virtual address for
+ *                  the region. Pass NULL to ignore.
+ * @param[out] size Optional pointer to receive the size of the MMIO region.
+ *                  Pass NULL to ignore.
+ * @returns Non-zero on error (including index out-of-range). Zero on success.
```
Comment (r-rojo) [inline include/opae/uio.h:121]: Is the idea here that you can use this to see what region indices are valid?

**Review by r-rojo — APPROVED** [review]

File: include/opae/uio.h (hunk 1 above)
Comment (tswhison) [reply include/opae/uio.h:29]: Right there is some specific knowledge of our sysfs layout which would preclude this from being used for any arbitrary device.

File: include/opae/uio.h (hunk 2 above)
Comment (tswhison) [reply include/opae/uio.h:121]: That would be a way to do that, yes.

**Pull request merged**